### ISSUE TITLE
Use LIMIT xxx with "WITH FILL " shows the expected results？

### ISSUE THREAD

**Issue by hexiaoting** [ClickHouse/ClickHouse#17466]
```
select version();

┌─version()─┐
│ 20.13.1.1 │
└───────────┘


### query1:
SELECT toFloat32(number % 10) AS n, 'original' AS source　   FROM numbers(10) WHERE number % 3 = 1;
┌─n─┬─source───┐
│ 1 │ original │
│ 4 │ original │
│ 7 │ original │
└───┴──────────┘

3 rows in set. Elapsed: 0.004 sec.

### query2:
SELECT toFloat32(number % 10) AS n, 'original' AS source　   FROM numbers(10) WHERE number % 3 = 1 order by n with fill step 1
┌─n─┬─source───┐
│ 1 │ original │
│ 2 │          │
│ 3 │          │
│ 4 │ original │
│ 5 │          │
│ 6 │          │
│ 7 │ original │
└───┴──────────┘

7 rows in set. Elapsed: 0.004 sec.


### query3:
SELECT toFloat32(number % 10) AS n, 'original' AS source　   FROM numbers(10) WHERE number % 3 = 1 order by n with fill step 1 limit 2;
┌─n─┬─source───┐
│ 1 │ original │
│ 2 │          │
│ 3 │          │
│ 4 │ original │
└───┴──────────┘

4 rows in set. Elapsed: 0.004 sec.

 SELECT toFloat32(number % 10) AS n, 'original' AS source　   FROM numbers(10) WHERE number % 3 = 1 order by n with fill step 1 limit 2 with ties;
┌─n─┬─source───┐
│ 1 │ original │
│ 2 │          │
└───┴──────────┘

2 rows in set. Elapsed: 0.004 sec.

The result returns 4 rows while the sql query use " limit 2" but return 2 rows when using "limit 2 with ties", Is this the expected behavior?
Is it a bug?
```

**Comment by hexiaoting** [comment]
@CurtizJ  is this a bug?

**Comment by CurtizJ** [comment]
I think yes. This inconsistency should be fixed.

**Comment by CurtizJ** [comment]
Fixed in #18188 #18516.
